x-pack/solutions/security/plugins/security_solution/server/endpoint/services/custom_scripts/clients/lib/base_custom_scripts_client.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { Logger } from '@kbn/logging';
+import type { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+import type { ActionsClient } from '@kbn/actions-plugin/server';
+import { CustomScriptsNotSupportedError } from './errors';
+import type { CustomScriptsResponse } from '../../../../../../common/endpoint/types/custom_scripts';
+import type { ResponseActionAgentType } from '../../../../../../common/endpoint/service/response_actions/constants';
+import type { EndpointAppContextService } from '../../../../endpoint_app_context_services';
+import type { CustomScriptsClientInterface } from './types';
+
+/**
+ * Options for creating a custom scripts client
+ */
+export interface CustomScriptsClientOptions {
+  endpointService: EndpointAppContextService;
+  esClient: ElasticsearchClient;
+  soClient: SavedObjectsClientContract;
+  connectorActionsClient?: ActionsClient;
+}
+
+/**
+ * Base class for all custom scripts clients
+ * Provides common functionality and enforces the interface
+ */
+export abstract class CustomScriptsClient implements CustomScriptsClientInterface {
```

Comment:
Why a new client? Would it make more sense to just add support to existing response action clients since this is part of response actions?

Reply:
Also - `AgentStatus` is different. Its not specific to response actions and is used through out the UI outside of response actions

Reply:
Sounds good! I can move it there, thanks :) 
Curious though - why is this different from Agent Status Client?

Reply:
😄 

> Re: "Why is this different from Agent Status Client?"

see my other comment above answering that specific question